src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/Utilities.cs

```diff
@@ -11,6 +15,30 @@ namespace Microsoft.DotNet.SourceBuild.SmokeTests;
 
 public static class Utilities
 {
+    public static void ExtractFileFromTarball(string tarballPath, string filePath, string outputDir)
+    {
+        Matcher matcher = new();
+        matcher.AddInclude(filePath);
+
+        using FileStream fileStream = File.OpenRead(tarballPath);
+        using GZipStream decompressorStream = new(fileStream, CompressionMode.Decompress);
+        using TarReader reader = new(decompressorStream);
```

Comment:
if we care to handle `.tar` files in addition to `.tar.gz` in the same method
```suggestion
        using TarReader reader = new(filePath.EndsWith(".gz", StringComparison.OrdinalIgnoreCase) ?
            new GZipStream(fileStream, CompressionMode.Decompress) : fileStream);
```